File: docs/src/main/scala/sexamples/virtualnetworking/pingpongselectors/IdChannelSelector.scala
package sexamples.virtualnetworking.pingpongselectors

import java.nio.ByteBuffer
import se.sics.kompics.ChannelSelector

class IdChannelSelector(id: Array[Byte])
    extends ChannelSelector[TMessage, ByteBuffer](classOf[TMessage], ByteBuffer.wrap(id), true) {
  override def getValue(event: TMessage): ByteBuffer = {
    event.header.dst.id match {
      case Some(eventId) => ByteBuffer.wrap(eventId)
      case None          => null
    }
  }
}

File: docs/src/main/java/jexamples/virtualnetworking/pingpongselectors/IdChannelSelector.java
package jexamples.virtualnetworking.pingpongselectors;

import java.nio.ByteBuffer;
import se.sics.kompics.ChannelSelector;

public class IdChannelSelector extends ChannelSelector<TMessage, ByteBuffer> {

  public IdChannelSelector(byte[] id) {
    super(TMessage.class, ByteBuffer.wrap(id), true);
  }

  @Override
  public ByteBuffer getValue(TMessage event) {
    return ByteBuffer.wrap(event.header.dst.getId());
  }
}
